solutions/usaco-380.mdx

```diff
@@ -131,5 +131,118 @@ public class CrossCountrySkiing {
 
 </JavaSection>
 
+<CPPSection>
+
+```cpp
+// @author: ascyrax
+
+#include <bits/stdc++.h>
+
+using namespace std;
+
+#define endl "\n"
+
+#define pb push_back
+
+int r, c;
+vector<vector<int>> grid;     //grid stores the elevation
```

Comment:
can you change this to a 2D array? it's more intuitive for storing a grid

Reply:
don't think it's more inuititve, vectors are generally better anyways

Reply:
I disagree, no reason to use a dynamic structure if an array suffices. 

Also, when I think "grid" I usually think 2D array, not a vectors of vectors (that seems more suited to like adj list)

Reply:
i can give you multiple links saying why c-style arrays are bad

Reply:
maybe someone else should comment their thoughts :)

no point arguing

Reply:
i agree with mikey

Reply:
the thing is arrays in general are just a pain to work with

Reply:
so what's the final decision. should I change it to a 2D array or leave it be

Reply:
it would be preferable if you changed it

Reply:
no, it really wouldn't